### PR TITLE
Fix-Location-Delete

### PR DESCRIPTION
**Pull request by MonishaSivanathan**
- fix location delete

### CODE REVIEW

File: controller/location_controller.py [hunk 1]
```diff
@@ -202,7 +202,8 @@ def update_location(resource_id: str, loc: LocationUpdateModel):
     @staticmethod
     def delete_location(resource_id: str, loc: LocationDeleteModel):
```
Comment (tamsud) [inline controller/location_controller.py:203]: If its delete API then it should do only the delete/soft delete.?
Why are we taking the status from the user?
Either modify the API to status_update/change the functionality to handle only delete (inactive/cancelled).

**Review by tamsud — CHANGES_REQUESTED** [review]

File: routes/location_routes.py [hunk 2]
```diff
@@ -63,12 +63,12 @@ async def update_location(
     logger.info(f"Update location {resource_id}")
     return LocationClient.update_location(resource_id, location)
 
-@router.delete("/location")
+@router.patch("/location")
 @permission_required("LOCATION", "DELETE")
```
Comment (tamsud) [inline routes/location_routes.py:67]: Using DELETE permission for an update operation is not good practice.

**Review by tamsud — CHANGES_REQUESTED** [review]

File: controller/location_controller.py (hunk 1 above)
Comment (MonishaSivanathan) [reply controller/location_controller.py]: changed the permission tamil

File: routes/location_routes.py (hunk 2 above)
Comment (MonishaSivanathan) [reply routes/location_routes.py]: changed into update @tamsud

**Review by tamsud — APPROVED** [review]

**Pull request merged**